### PR TITLE
Implements a routine for Index Lambda to High Level Op

### PR DESCRIPTION
**Pull request by kaushikcfd**
Q. Why is this useful?
A. Realizing how an IndexLambda came into existence is important for several reasons:
- Implementing metadata translation
- Allowing pattern matching on sub-expressions while exposing the high-level operations.
  - Raising to HLO also allows for richer pattern matching as we can match based on properties of the high level ops (like commutativity, associativity) 
- Implementing a Numpy-like target
- Implementing a smart `Array.__str__`.

### CODE REVIEW

File: doc/array.rst [hunk 1]
```diff
@@ -5,6 +5,12 @@ Array Expressions
 
 .. automodule:: pytato.array
 
+
+Raising :class:`~pytato.array.IndexLambda` nodes
+------------------------------------------------
+
+.. automodule:: pytato.normalization
```
Comment (inducer) [inline doc/array.rst:12]: Doc nomenclature and module name clash. Is it "normalization" or "raising"?

**Review by inducer — APPROVED** [review]
Thanks! This LGTM other than a quibble about naming.

File: doc/array.rst (hunk 1 above)
Comment (kaushikcfd) [reply doc/array.rst:12]: I think "raising" makes more sense. I'll quickly change this.

Comment (kaushikcfd) [reply doc/array.rst]: Thanks, done: https://github.com/inducer/pytato/compare/0a8fb7ae1a0941c69ff1da12f7efbc80fb276de1..e113cc16d2b8b56763e47150cdee4fddeda8b85c

**Comment by inducer** [comment]
LGTM. In it goes!

**Pull request merged**